Store `chunk_domain_size` explicitly in `Chunk`.

nnethercote opened the pull request:
<!-- homu-ignore:start -->
*[View all comments](https://triagebot.infra.rust-lang.org/gh-comments/rust-lang/rust/pull/147802)*
<!-- homu-ignore:end -->

Currently we compute it on demand, but it's a little simpler and slightly faster to store it.

r? @ghost

nnethercote commented:
@bors try @rust-timer queue

rust-timer commented:
Awaiting bors try build completion.

@rustbot label: +S-waiting-on-perf

<!-- rust-timer: temporary -->

nnethercote committed:
Store `chunk_domain_size` explicitly in `Chunk`.

Currently we compute it on demand, but it's a little simpler and
slightly faster to store it.

nnethercote commented:
@bors try @rust-timer queue

rust-timer commented:
Queued bd78cbb7e34d44cad00846b8f105b56cc2de5222 with parent bcded331651b60a0383b3ff51db4f24c4495ac53, future [comparison URL](https://perf.rust-lang.org/compare.html?start=bcded331651b60a0383b3ff51db4f24c4495ac53&end=bd78cbb7e34d44cad00846b8f105b56cc2de5222).
There are currently 0 preceding artifacts in the [queue](https://perf.rust-lang.org/status.html).
It will probably take at least ~1.0 hours until the benchmark run finishes.
<!-- rust-timer: temporary -->

rust-timer commented:
Finished benchmarking commit (bd78cbb7e34d44cad00846b8f105b56cc2de5222): [comparison URL](https://perf.rust-lang.org/compare.html?start=bcded331651b60a0383b3ff51db4f24c4495ac53&end=bd78cbb7e34d44cad00846b8f105b56cc2de5222&stat=instructions:u).

### Overall result: ✅ improvements - no action needed


Benchmarking this pull request means it may be perf-sensitive – we'll automatically label it not fit for rolling up. You can override this, but we strongly advise not to, due to possible changes in compiler perf.

@bors rollup=never
@rustbot label: -S-waiting-on-perf -perf-regression


### [Instruction count](https://perf.rust-lang.org/compare.html?start=bcded331651b60a0383b3ff51db4f24c4495ac53&end=bd78cbb7e34d44cad00846b8f105b56cc2de5222&stat=instructions:u)
Our most reliable metric. Used to determine the overall result above. However, even this metric can be noisy.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -0.8% | [-1.2%, -0.4%] | 4     |
| Improvements ✅ <br /> (secondary) | -     | -              | 0     |
| All ❌✅ (primary)                 | -0.8% | [-1.2%, -0.4%] | 4     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=bcded331651b60a0383b3ff51db4f24c4495ac53&end=bd78cbb7e34d44cad00846b8f105b56cc2de5222&stat=max-rss)
<details>
<summary>Results (secondary -5.1%)</summary>

A less reliable metric. May be of interest, but not used to determine the overall result above.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -5.1% | [-5.1%, -5.1%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=bcded331651b60a0383b3ff51db4f24c4495ac53&end=bd78cbb7e34d44cad00846b8f105b56cc2de5222&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.

### [Binary size](https://perf.rust-lang.org/compare.html?start=bcded331651b60a0383b3ff51db4f24c4495ac53&end=bd78cbb7e34d44cad00846b8f105b56cc2de5222&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 486.041s -> 487.115s (0.22%)
**Artifact size**: 395.29 MiB -> 395.25 MiB (-0.01%)

nnethercote commented:
I rebased, and added a brief comment about the history. Perf is still improved on cranelift-codegen.

@bors r=Mark-Simulacrum

panstromek commented:
perf triage:

`ctfe-stress-5` regression is a noise spike.

@rustbot label: +perf-regression-triaged
